openmdao/core/system.py

```diff
@@ -2940,7 +2940,10 @@ def get_design_vars(self, recurse=True, get_sizes=True, use_prom_ivc=True):
 
                 if 'size' not in meta:
                     if src_name in abs2idx:
-                        meta['size'] = sizes[owning_rank[src_name], abs2idx[src_name]]
+                        if out[abs_name]['distributed']:
+                            meta['size'] = sizes[self.comm.rank, abs2idx[src_name]]
```

Comment:
For a distributed variable, you should just be able to set `meta['size'] = abs2meta[name]['global_size']`.

Reply:
Global size doesn't own the size at that rank. Eg. three procs might have values of `3,2,2` respectively and global size holds `7` in that case. Also, not all abs2meta has `'size'`.

Reply:
I was thinking that because drivers (at least all of our current ones) use the full distributed variable instead of just the local part of it, that the 'size' of a design var/response would just be set to the global size, but it looks like we label size and global_size separately for desvars/responses just like we do with input/output vector metadata.  I guess that makes sense because maybe there could be some sort of distributed driver someday that only used the local part.  I'm curious where you saw an abs2meta entry that didn't have 'size'.  They should all have that entry since abs2meta doesn't contain any discrete variables.

Reply:
You were right, I was using `name` when I should've been using `src_name`. Changed